python/revng/api/_capi.py

```diff
@@ -0,0 +1,64 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import os
+import traceback
+from pathlib import Path
+from typing import List
+
+from cffi import FFI
+from cffi.backend_ctypes import CTypesBackend
+
+# We assume this file is in <root>/lib/python/revng/api
+ROOT = (Path(__file__) / "../../../../..").resolve()
+HEADERS_DIR = ROOT / "include" / "revng" / "PipelineC"
+
+header_paths = [
+    HEADERS_DIR / "ForwardDeclarationsC.h",
+    HEADERS_DIR / "Prototypes.h",
+]
+
+ffi = FFI()
+for header_path in header_paths:
+    with open(header_path, encoding="utf-8") as header_file:
+        lines = []
+        for line in header_file:
+            if not line.startswith("#"):
+                lines.append(line)
+        ffi.cdef("\n".join(lines))
+
+
+@ffi.callback("void ()")
+def pytraceback():
+    traceback.print_stack()
+
+
+LIBRARY_PATH = ROOT / "lib" / "librevngPipelineC.so"
+assert LIBRARY_PATH.is_file()
+
+ARGV: List[str] = []
+
+REVNG_ANALYSIS_LIBRARIES = os.getenv("REVNG_ANALYSIS_LIBRARIES", "")
+ANALYSIS_LIBRARIES = [Path(f) for f in REVNG_ANALYSIS_LIBRARIES.split(":")]
+
+# TODO: this needs to go in /lib/analyses
+ANALYSIS_LIBRARIES.append(ROOT / "lib/librevngRecompile.so")
```

Comment:
Fix the library, we never ever merge these hacks.

Reply:
Fixed